autogen/sr.rs

```diff
@@ -299,14 +306,23 @@ pub fn gen_sr_code_from_instruction_grammar(
 
         match inst.class {
             Some(structs::Class::Type) => {
-                type_variants.push(if field_names.is_empty() {
-                    quote!{ #type_ident }
+                if field_names.is_empty() {
+                    type_variants.push(quote!{ #type_ident });
+                    type_lifts.push(quote! {
+                        #opcode => Ok(Type::#type_ident),
+                    });
                 } else {
-                    quote! { #type_ident {
-                        #( #field_names: #field_types ),*
-                    }}
-                });
-
+                    type_variants.push(quote! {
+                        #type_ident {
+                            #( #field_names: #field_types ),*
+                        }
+                    });
+                    type_lifts.push(quote! {
+                        #opcode => Ok(Type::#type_ident {
+                            #( #field_names: #field_lifts, )*
+                        }),
+                    });
+                }
```

Comment:
Hmmm, I've seen this exact code before... :grin: It shouldn't be abstracted yet but it's something to keep an eye on.

Reply:
yes, absolutely!